test/test_common/network_utility.cc

```diff
@@ -1,22 +1,87 @@
 #include "network_utility.h"
 
+#include "envoy/common/exception.h"
+#include "common/common/assert.h"
+#include "common/network/address_impl.h"
+#include "utility.h"
+
+//#include <errno.h>
+
 namespace Network {
 namespace Test {
-namespace {
-const int kFirstEphemeralPort = 40152;
-const int kLastEphemeralPort = 65535;
-const int kNumEphemeralPorts = kLastEphemeralPort - kFirstEphemeralPort + 1;
-} // namespace
 
-int getUnusedPort() {
-  static int next_port = kFirstEphemeralPort;
-  static std::mutex m;
-  std::lock_guard<std::mutex> lock(m);
-  if (next_port > kLastEphemeralPort) {
-    next_port = kFirstEphemeralPort;
+Address::InstancePtr checkPortAvailability(Address::InstancePtr addrPort,
+                                           Address::SocketType type) {
+  // Note that there isn't any point in passing in a port of zero as we don't provide
+  // a means of finding out what port the OS assigned in that case. That could be fixed
+  // if desired.
+  if (addrPort == nullptr) {
+    // TODO(jamessynge) Add logging of err since it isn't expected, for now throwing.
+    throw EnvoyException("addrPort == nullptr");
```

Comment:
Why not use an ASSERT? This is test code. One less place for explicit exceptions.